theano/tensor/opt.py

```diff
@@ -1457,8 +1457,10 @@ def apply(self, fgraph):
 
 # Register it after merge1 optimization at 0. We don't want to track
 # the shape of merged node.
+# As local_shape_to_shape_i is not in fast_compile,
+# dont register this opt in fast_compile.
 theano.compile.mode.optdb.register('ShapeOpt', ShapeOptimizer(),
-                                   0.1, 'fast_run', 'fast_compile')
+                                   0.1, 'fast_run')
```

Comment:
Try this diff:

diff --git a/theano/tensor/opt.py b/theano/tensor/opt.py
index 64c0478..fc80dd2 100644
--- a/theano/tensor/opt.py
+++ b/theano/tensor/opt.py
@@ -3310,7 +3310,8 @@ def local_adv_sub1_adv_inc_sub1(node):
             T.extract_constant(x, elemwise=False) != 0):
         return
     cond = [T.all(T.and_(T.lt(idx, x.shape[0]), T.ge(idx, -x.shape[0])))]
-    if not node.fgraph.shape_feature.same_shape(idx, y, 0, 0):
+    if (not hasattr(node.fgraph, 'shape_feature') or
+            not node.fgraph.shape_feature.same_shape(idx, y, 0, 0)):
         cond.append(T.eq(idx.shape[0], y.shape[0]))
     r = Assert("Bad indexing or shapes in a AdvancedIncSubtensor1 "
                "that was optimized away")(y, *cond)